### PR TITLE
Longest increasing subsequence added

### PR DESCRIPTION
**Pull request by sHiVaNgI821**
Added the longest_increasing_subsequence method that returns the longest strictly increasing subsequence that can be obtained from the given array. The lower_bound function used in the implementation has been taken from #351

### CODE REVIEW

**Comment by Smit-create** [comment]
@sHiVaNgI821 Please resolve merge conflicts since your previous PR has merged.

**Comment by Smit-create** [comment]
@czgdp1807 Please have a look

**Comment by czgdp1807** [comment]
Seems good. Would prefer to use, snake case instead of camel case in the doc string examples.

**Comment by Smit-create** [comment]
Looks ready to be merged.

File: pydatastructs/linear_data_structures/algorithms.py [hunk 1]
```diff
@@ -978,3 +979,62 @@ def lower_bound(array, value, **kwargs):
...
+def longest_increasing_subsequence(array):
+    """
+    Returns the longest increasing subsequence (as a OneDimensionalArray) that
+    can be obtained from a given OneDimensionalArray. A subsequence
+    of an array is an ordered subset of the array's elements having the same
+    sequential ordering as the original array. Here, an increasing
+    sequence stands for a strictly increasing sequence of numbers.
+
+    Parameters
+    ========
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/algorithms.py:992]:
```suggestion
    Parameters
    =========
```

**Comment by czgdp1807** [comment]
Will merge once the tests pass.

**Comment by czgdp1807** [comment]
Thanks.

**Pull request merged**